pkg/kubectl/cmd/label.go

```diff
@@ -0,0 +1,167 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/meta"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/spf13/cobra"
+)
+
+func (f *Factory) NewCmdLabel(out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "label [--overwrite] <resource> <name> <key-1>=<val-1> ... <key-n>=<val-n> [--resource-version=<version>]",
+		Short: "Update the labels on a resource",
+		Long: `Update the labels on a resource.
+
+If --overwrite is true, then existing labels can be overwritten, otherwise attempting to overwrite a label will result in an error.
+If --resource-version is specified, then updates will use this resource version, otherwise the existing resource-version will be used.
+
+Examples:
+  $ kubectl label pods foo unhealthy=true
+  <update a pod with the label 'unhealthy' and the value 'true'>
+
+  $ kubectl label --overwrite pods foo status=unhealthy
+  <update a pod with the label 'status' and the value 'unhealthy' overwritting an existing value>
+  
+  $ kubectl label pods foo status=unhealthy --resource-version=1
+  <update a pod with the label 'status' and the value 'unhealthy' if the resource is unchanged from version 1>`,
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) < 2 {
+				usageError(cmd, "<resource> <name> is required")
+			}
+			if len(args) < 3 {
+				usageError(cmd, "at least one label update is required.")
+			}
+			res := args[:2]
+			cmdNamespace, err := f.DefaultNamespace(cmd)
+			checkErr(err)
+
+			mapper, _ := f.Object(cmd)
+			mapping, namespace, name := util.ResourceFromArgs(cmd, res, mapper, cmdNamespace)
+			client, err := f.RESTClient(cmd, mapping)
+			checkErr(err)
+
+			labels, remove, err := parseLabels(args[2:])
+			checkErr(err)
+			overwrite := util.GetFlagBool(cmd, "overwrite")
+			resourceVersion := util.GetFlagString(cmd, "resource-version")
+
+			obj, err := updateObject(client, mapping, namespace, name, func(obj runtime.Object) runtime.Object {
```

Comment:
I didn't see a response to the bulk-update issue.

Example is here:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubectl/cmd/update.go#L57

Obviously, a resource-version precondition would be incompatible with a bulk update.